src/main/thrift/parquet.thrift

```diff
@@ -481,9 +481,10 @@ enum CompressionCodec {
   SNAPPY = 1;
   GZIP = 2;
   LZO = 3;
-  BROTLI = 4; // Added in 2.4
-  LZ4 = 5;    // Added in 2.4
-  ZSTD = 6;   // Added in 2.4
+  BROTLI = 4;  // Added in 2.4
+  LZ4 = 5;     // DEPRECATED (Added in 2.4)
+  ZSTD = 6;    // Added in 2.4
+  LZ4_RAW = 7; // Added in 2.9
```

Comment:
I don't think we should be too hasty to add LZ4 back into the spec, or at least we should narrow the scope.  The integration of LZ4 with arrow has exposed problematic java implementations of LZ4: https://github.com/apache/arrow/pull/8949#issuecomment-794124675

Reply:
This is proposing the LZ4 _block_ format, not the LZ4 _frame_ format which the dependent blocks error seems to be stemming from.

Reply:
Hmm, OK, as long as we confirm that consumption can happen bidirectionally we the java library I'm happy :)

Reply:
It certainly can theoretically happen. It's not the task of this PR to write an implementation, though.